subprojects/core/src/main/java/org/gradle/process/internal/streams/ExecOutputHandleRunner.java

```diff
@@ -62,7 +63,7 @@ public void run() {
     }
 
     public void run(Executor executor) {
-        executor.execute(this);
+        executor.execute(new BuildOperationIdentifierPreservingRunnable(this));
```

Comment:
I didn't see any other places that were using this run method.  I would probably get rid of it and decorate things in StreamsForwarder.